### DISCUSSION TITLE
Add better support for customising backoffice login screen.

### DISCUSSION THREAD

**Discussion by shropgov-neil** [umbraco/Umbraco-CMS#15779] [Features and ideas]
The support for customising the backoffice login screen is very limited, as documented.

It appears that any CSS beyond the explictly documented variables is either removed or intentionally overriden, which stops any additional customisation such as tweaks to meet the WCAG 2.1 recomendations, or alignment/position/sizing issues with the image wrapper/login wrapper.

Additionaly there is no supported way to modify the markup (at least, none that I can find in the docs) - for example I'd like to re-introduce the URL into the title tag (that was removed in v13), as it is needed for my password manager to find the correct record. I'd also like to add a warning about unauthorised access.

Is there a concious design decision to limit developer control over this page (and why), or is it something that can be allowed in a future release?

**Comment by skttl** [comment]
Probably not the right solution, but you can add a copy of [this file](https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Cms.StaticAssets/umbraco/UmbracoLogin/Index.cshtml) in /umbraco/UmbracoLogin/Index.cshtml, and then it will override the built in one. Then you can do all your modifications there - with the caveat, that you need to merge in future changes manually.

Besides that, maybe we can improve the built in login screen? Better accessibility sounds like something that all would benefit from eg.

**Reply by shropgov-neil** [reply]
cheers, I did think about doing something like that - but it felt a little clunky, especially just for the CSS overrides.

Not sure what the alternative would be for DOM changes, however.

**Reply by joecklau** [reply]
> Probably not the right solution, but you can add a copy of [this file](https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Cms.StaticAssets/umbraco/UmbracoLogin/Index.cshtml) in /umbraco/UmbracoLogin/Index.cshtml, and then it will override the built in one. Then you can do all your modifications there - with the caveat, that you need to merge in future changes manually.
> 
> Besides that, maybe we can improve the built in login screen? Better accessibility sounds like something that all would benefit from eg.

Is there similar thing in v10? I also need to replace the login screen DOM ...orz

**Reply by joecklau** [reply]
> > Probably not the right solution, but you can add a copy of [this file](https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Cms.StaticAssets/umbraco/UmbracoLogin/Index.cshtml) in /umbraco/UmbracoLogin/Index.cshtml, and then it will override the built in one. Then you can do all your modifications there - with the caveat, that you need to merge in future changes manually.
> > Besides that, maybe we can improve the built in login screen? Better accessibility sounds like something that all would benefit from eg.
> 
> Is there similar thing in v10? I also need to replace the login screen DOM ...orz

Self-answering, the file for v10 is under Umbraco.Cms.StaticAssets\umbraco\UmbracoBackOffice\Default.cshtml, copying to \umbraco\UmbracoBackOffice\Default.cshtml and customize on it works
